libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -626,17 +626,23 @@ void AP_GPS::update_instance(uint8_t instance)
     // detection to run again
     if (!result) {
         if (tnow - timing[instance].last_message_time_ms > GPS_TIMEOUT_MS) {
-            // free the driver before we run the next detection, so we
-            // don't end up with two allocated at any time
-            delete drivers[instance];
-            drivers[instance] = nullptr;
-            memset(&state[instance], 0, sizeof(state[instance]));
-            state[instance].instance = instance;
-            state[instance].status = NO_GPS;
-            state[instance].hdop = GPS_UNKNOWN_DOP;
-            state[instance].vdop = GPS_UNKNOWN_DOP;
-            timing[instance].last_message_time_ms = tnow;
-            timing[instance].delta_time_ms = GPS_TIMEOUT_MS;
+            // do not try to detect again if type is MAV
+            if (_type[instance] == GPS_TYPE_MAV) {
+                state[instance].status = NO_FIX;
```

Comment:
Rather then selectively filling this out here, you are probably better off memsetting state first before the if (as well as set the DOP), then inside the if all you have to do is select the status between NO_FIX and NO_GPS. (As well as do the driver delete in the else case).